js/WeatherApp.js

```diff
@@ -0,0 +1,241 @@
+import icons from './icons.js';
+import background from './background.js';
+import windDirection from './wind.js';
+
+class WeatherApp {
+	constructor() {
+		this.container = document.querySelector('.container');
+		this.apiKey = 'c00b4b29cd8af0576010b2bbfd606608';
+		this.assingElements();
+		this.getGeolocation();
+		this.addEvents();
+	}
+
+	assingElements() {
+		this.input = this.container.querySelector('#city');
+		this.form = this.container.querySelector('form');
+		this.mainBox = this.container.querySelector('main');
+		this.errorMessage = this.container.querySelector('.error-message');
+		this.infoApp = this.container.querySelector('.info-app');
+
+		this.input.focus();
+	}
+
+	addEvents() {
+		this.form.addEventListener('submit', this.getWeather.bind(this));
+	}
+
+	async getGeolocation() {
+		if (navigator.geolocation) {
+			try {
+				const position = await this.geolocateUser();
+				const lat = position.coords.latitude;
+				const lon = position.coords.longitude;
+				this.url = `https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&appid=${this.apiKey}`;
+				this.getWeather();
+
+			} catch (e) {
+				console.error(e.message, e);
+			}
+		}
+	}
+
+	geolocateUser() {
+		return new Promise((resolve, reject) => navigator.geolocation.getCurrentPosition(resolve, reject));
+	}
+
+	getWeather(e) {
+		e?.preventDefault();
+
+		if (this.input.value.trim() == '') {
+			this.displayErrorMessage('Try writting something 😅');
+		} else {
+			const city = this.input.value;
+			this.url = `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${this.apiKey}`;
+			this.consultAPI();
+		}
+	};
+
+	consultAPI() {
+		// API Documentation --> https://openweathermap.org/weather-data
+		fetch(this.url)
+			.then((response) => {
+				if (response.status != 200) {
+					throw response.statusText;
+				}
+				return response.json();
+			})
+			.then(data => {
+				this.weatherData = data;
+				this.resetInput();
+				this.displayData();
+			})
+			.catch(e => {
+				console.error(e);
+				this.displayErrorMessage(`Sorry, ${e.toLowerCase()}, please try again 💚`);
+			});
+	}
+
+	displayErrorMessage(message) {
+		this.resetAll();
+		this.errorMessage.innerHTML = message;
+	}
+
+	resetAll() {
+		this.resetBackground();
+		this.resetInput();
+		this.resetData();
+	}
+
+	resetData() {
+		this.infoApp.innerHTML = '';
+	}
+
+	resetInput() {
+		this.input.value = '';
+		this.input.focus();
+	}
+
+	resetErrorField() {
+		this.mainBox.style.height = 'auto';
+		this.errorMessage.innerHTML = '';
+	}
+
+	displayData() {
+		this.resetErrorField();
+		this.parseAPIData();
+		this.createTemplate();
+
+		this.input.setAttribute("placeholder", "try another city");
+	};
```

Comment:
```suggestion
	displayData() {
		this.resetErrorField();
		this.parseAPIData();
		this.getBackground();
		this.createTemplate();

		this.input.setAttribute("placeholder", "try another city");
	};
```

Reply:
Ahora que lo estoy releyendo, no deberían llamarse `this.setBackground()`?? Lo mismo para todas las demás `get`

Reply:
Si! ya lo cambio!